bin/validate_dn_format.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+
+import glob
+import os
+import re
+
+regex_dn = re.compile(r"^(/(?:DC|OU|C|ST|L|O)=[^/=]+)*(/CN=([^/=].+))$")
+
+error_log = []
+
+whitelist = [
+    'voms-alice-auth.app.cern.ch',
+    'voms-atlas-auth.app.cern.ch',
+    'voms-cms-auth.app.cern.ch',
+    'voms-lhcb-auth.app.cern.ch',
+    'voms-ops-auth.app.cern.ch',
+]
+
+def validate(dir):
+    for file_path in glob.glob(dir + "/*/*.lsc"):
+        with open(file_path, 'r') as f:  # reads .lsc files line by line
+            lines = f.read().splitlines()
+
+            if check_number_of_dn(lines):
+                error_log.append(f'Error in "{file_path}" .lsc file should contain exactly 2 DNs')
+            
+            if check_matching_cn(lines[0], file_path):
+                error_log.append(f'Error in "{file_path}" at "{lines[0]}" subject CN value does not match the file name')
+
+            for line in lines:
+                if check_format(line):
+                    error_log.append(f'Error in "{file_path}" at "{line}" invalid format')
```

Comment:
```suggestion
            if not check_number_of_dn(lines):
                error_log.append(f'Error in "{file_path}" .lsc file should contain exactly 2 DNs')
            
            if not check_matching_cn(lines[0], file_path):
                error_log.append(f'Error in "{file_path}" at "{lines[0]}" subject CN value does not match the file name')

            for bad_line in [line for lines if not check_format(line)]:
                error_log.append(f'Error in "{file_path}" at "{bad_line}" invalid format')
```

Dealing with the swapped logic and using a list comprehension instead. Although it shouldn't really matter in this case since our lists will be small, filtering with a list comprehension is faster than a separate for loop + if block!

Reply:
Got it. Will keep this in mind for future scripts.